gemato/openpgp.py

```diff
@@ -588,23 +588,46 @@ def refresh_keys_wkd(self):
             data,
             raise_on_error=OpenPGPKeyRefreshError)
 
-        # we need to explicitly ensure all keys were fetched
+        # Need to explicitly ensure all keys were fetched
+        # However:
+        # - any key MAY appear 0 or more times.
+        # - expected keys SHOULD be present.
+        # - unexpected keys MAY also be present.
+        #
+        # Track updated & unexpected keys to compare vs expected keys.
+        updated_keys = set()
+        unexpected_keys = set()
```

Comment:
At this point, I'd put all keys printed by GPG into a single set, and then use set operations to consider "updated" and "unexpected" keys.

Reply:
done